search/intermediate_result.cpp

```diff
@@ -282,19 +285,38 @@ void FillDetails(FeatureType & ft, Result::Details & details)
 
   feature::TypesHolder const typesHolder(ft);
 
-  details.m_isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
-  if (details.m_isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), details.m_stars))
-    details.m_stars = std::min(details.m_stars, osm::MapObject::kMaxStarsCount);
-  else
-    details.m_stars = 0;
+  uint8_t starsCount = 0;
+  bool isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
+  if (isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), starsCount))
+    starsCount = std::min(starsCount, osm::MapObject::kMaxStarsCount);
+  std::string stars;
+  for (int i = 0; i < starsCount; ++i)
+    stars.append(osm::MapObject::kStarSymbol);
 
   auto const cuisines = feature::GetLocalizedCuisines(typesHolder);
-  details.m_cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
+  std::string cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
```

Comment:
1. All non-modified vars should be const.
2. `auto const` is clear enough.

Reply:
Done